### ISSUE TITLE
Read byte[] depends on snappy

### ISSUE THREAD

**Issue by devinrsmith** [OpenHFT/Chronicle-Wire#30]
I'm able to write bytes with chronicle queue. Using a TextWire, it looks like things get pushed out base64 encoded and uncompressed.

``` java
    @Override
    public void writeMarshallable(@NotNull WireOut wire) {
        wire.getValueOut().bytes(bytes);
    }
```

I'm trying to implement the read side of things:

``` java
    @Override
    public void readMarshallable(@NotNull WireIn wire) throws IORuntimeException {
        System.arraycopy(wire.getValueIn().bytes(), 0, bytes, 0, bytes.length);
    }
```

Unfortunately, it can't load up `Compressions`:

```
java.lang.NoClassDefFoundError: org/xerial/snappy/SnappyInputStream
    at net.openhft.chronicle.bytes.util.Compressions.<clinit>(Compressions.java:101)
    at net.openhft.chronicle.bytes.util.Compression.uncompress(Compression.java:99)
```

Am I using the interface incorrectly?

I could import the snappy library, but it's a bit inefficient since we are only using `byte[]`s for small ids that don't need any sort of snappy compression.

**Comment by peter-lawrey** [comment]
Yes. This doesn't look very efficient though you could get it to work. Let
me get back to you with an example.

Why are you using Marshallable with a byte [] ?

When you dump the byte [] as text it should show it encoded.

**Comment by devinrsmith** [comment]
Agreed, this isn't very efficient. My actual impl doesn't create garbage:

```java
    @Override
    public void readMarshallable(@NotNull WireIn wire) throws IORuntimeException {
        wire.getValueIn().bytes(this);
    }

    @Override
    public void readMarshallable(BytesIn bytes) {
        bytes.read(this.bytes);
    }
```

but figured the `wire.getValueIn().bytes()` is slightly easier to use as an example.

I'm using `Marshallable` b/c it presents a more general / easy to use interface, especially as I compose different `Marshallable`s together in a single class. The (de)serialization simply involves iterating over all the `Marshallable`s, and I don't have to special case anything.

**Comment by peter-lawrey** [comment]
Can you provide a full stacktrace for the latest code? The line numbers have changed quite a bit and I can't see why it is trying to use snappy in the first place.

**Comment by devinrsmith** [comment]
```
java.lang.NoClassDefFoundError: org/xerial/snappy/SnappyOutputStream

	at net.openhft.chronicle.bytes.util.Compressions.<clinit>(Compressions.java:103)
	at net.openhft.chronicle.bytes.util.Compression.uncompress(Compression.java:100)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.bytes(TextWire.java:1747)
```

**Comment by devinrsmith** [comment]
```
java.lang.NoClassDefFoundError: org/xerial/snappy/SnappyOutputStream

	at net.openhft.chronicle.bytes.util.Compressions.<clinit>(Compressions.java:103)
	at net.openhft.chronicle.bytes.util.Compression.uncompress(Compression.java:100)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.bytes(TextWire.java:1789)
```
as of 
```
          <groupId>net.openhft</groupId>
          <artifactId>chronicle-queue</artifactId>
          <version>4.5.16</version>
```

**Comment by peter-lawrey** [comment]
That should be fixed in the latest snapshot. Will leave this open until we
can confirm this is the case.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
